File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

vector<int> days;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    for(int i = 0; i<progresses.size(); i++){
        days.push_back(ceil( (double)(100-progresses[i] ) / speeds[i]));
        // cout << days[i] << " ";
    }
    
    int max_before = days[0];
    int cnt = 1;
    
    for(int i = 1; i<days.size(); i++){
        if(max_before >= days[i]) cnt++;
        
        if(max_before < days[i]){
            answer.push_back(cnt);
            cnt = 1;
            max_before = days[i];
        }
        
        if(i == days.size()-1) answer.push_back(cnt);
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    let days = [];
    for(let i = 0; i<progresses.length; i++){
        let day = parseInt((100-progresses[i]) / speeds[i]);
        if((100-progresses[i]) % speeds[i] !== 0) day+= 1;
        days.push(day);
    }
    
    let now = days[0];
    let sum = 1;
    for(let i = 1; i<days.length; i++){
        if(days[i] <= now) sum++;
        if(days[i] > now){
            answer.push(sum);
            now = days[i];
            sum = 1;
        }
        
        if(i == days.length-1) answer.push(sum);
    }
    
    return answer;
}